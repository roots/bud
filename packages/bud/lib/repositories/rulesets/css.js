"use strict";
exports.__esModule = true;
exports.css = void 0;
var css = function (bud) {
    return bud.hooks.filter('webpack.module.rules.css', {
        test: bud.hooks.filter('webpack.module.rules.css.test', bud.patterns.get('css')),
        exclude: bud.hooks.filter('webpack.module.rules.css.exclude', bud.patterns.get('vendor')),
        use: bud.hooks.filter('webpack.module.rules.css.use', [
            bud.inProduction
                ? bud.uses.get('miniCss')(bud)
                : bud.loaders.get('style'),
            bud.uses.get('css')(bud),
            bud.uses.get('resolveUrl')(bud),
            bud.uses.get('postCss')(bud),
        ])
    });
};
exports.css = css;
//# sourceMappingURL=css.js.map