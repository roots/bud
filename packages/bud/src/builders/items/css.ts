export const ident: Framework.Item['ident'] = 'css'
export const loader: Framework.Item['loader'] = 'css-loader'
export const options: Framework.Item['options'] = {
  importLoaders: 2,
}
