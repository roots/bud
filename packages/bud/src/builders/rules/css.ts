export const test: Conditional = ({patterns}) =>
  patterns.get('css')

export const exclude: Exclude = ({patterns}) =>
  patterns.get('modules')

export const use: Use = ({build, mode}) => {
  const use: UseItem = item => build.getItem(item)
  const style = mode.is('production')
    ? use('minicss')
    : use('style')

  return [
    use('thread'),
    use('cache'),
    style,
    use('css'),
    use('resolveUrl'),
  ]
}

declare type Conditional = Framework.Rule.Factory<
  Framework.Rule.Conditional
>

declare type Exclude = Framework.Rule.Factory<
  Framework.Rule.Conditional
>

declare type Use = Framework.Rule.Factory<Framework.Rule.Use>
declare type UseItem = (item: string) => Framework.Rule.Generic
