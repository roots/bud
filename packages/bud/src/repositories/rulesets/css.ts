import type {Bud} from '../..'
import type {WebpackRule} from '@roots/bud-typings'

const css = (bud: Bud): WebpackRule =>
  bud.hooks.filter('webpack.module.rules.css', {
    test: bud.hooks.filter(
      'webpack.module.rules.css.test',
      bud.patterns.get('css'),
    ),
    exclude: bud.hooks.filter(
      'webpack.module.rules.css.exclude',
      bud.patterns.get('vendor'),
    ),
    use: bud.hooks.filter('webpack.module.rules.css.use', [
      bud.inProduction
        ? bud.uses.get('miniCss')(bud)
        : bud.loaders.get('style'),
      bud.uses.get('css')(bud),
      bud.uses.get('resolveUrl')(bud),
      bud.uses.get('postCss')(bud),
    ]),
  })

export {css}
