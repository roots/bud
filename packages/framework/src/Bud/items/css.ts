export const ident: Build.Item['ident'] = 'css'

export const loader: Build.Item['loader'] = 'css'

export const options: Build.Item['options'] = {
  importLoaders: 1,
}
