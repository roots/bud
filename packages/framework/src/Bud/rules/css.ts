export const test: Conditional = bud =>
  bud.store['patterns'].get('css')

export const exclude: Exclude = bud =>
  bud.store['patterns'].get('modules')

export const use: Build.Rule.Factory<Build.Rule.Use> = bud => {
  const use: UseLoader = item => bud.build.items[item].make()

  return [
    bud.mode.is('production') ? use('minicss') : use('style'),
    use('css'),
    use('resolveUrl'),
  ]
}

declare type Conditional = Build.Rule.Factory<
  Build.Rule.Conditional
>

declare type Exclude = Build.Rule.Factory<Build.Rule.Conditional>

declare type UseLoader = (item: string) => Build.Rule.Generic
