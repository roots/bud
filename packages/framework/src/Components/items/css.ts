export const ident: Build.Item['ident'] = 'css-loader'

export const loader: Build.Item['loader'] = loaders =>
  loaders.get('css-loader')

export const options: Build.Item['options'] = {
  importLoaders: 1,
}
