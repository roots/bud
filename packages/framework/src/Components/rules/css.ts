export const test: Conditional = bud =>
  bud.store['patterns'].has('css')

export const exclude: Exclude = bud =>
  bud.store['patterns'].get('modules')

export const use: Build.Rule.Factory<Build.Rule.Use> = bud => {
  const use: Loaders.UseLoader = loader =>
    bud.components['items'].get(loader).make()

  return [
    bud.mode.is('production')
      ? use('minicss-loader')
      : use('style-loader'),
    use('css-loader'),
    use('resolve-url-loader'),
  ]
}

declare type Conditional = Build.Rule.Factory<
  Build.Rule.Conditional
>

declare type Exclude = Build.Rule.Factory<Build.Rule.Conditional>

declare namespace Loaders {
  export type UseLoader = (loader: string) => Build.Rule.Product
}
