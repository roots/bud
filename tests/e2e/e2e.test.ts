import {describe} from '@jest/globals'

import * as css from './e2e.css'
import * as js from './e2e.script'

describe('css', css.test)
describe('js', js.test)
