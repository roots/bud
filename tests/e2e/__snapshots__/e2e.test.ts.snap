// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`css has expected baseline: rgb(88, 19, 213) none repeat scroll 0% 0% / auto padding-box border-box 1`] = `"rgb(88, 19, 213) none repeat scroll 0% 0% / auto padding-box border-box"`;

exports[`css hot update css: rgb(0, 0, 0) none repeat scroll 0% 0% / auto padding-box border-box 1`] = `"rgb(0, 0, 0) none repeat scroll 0% 0% / auto padding-box border-box"`;
